### PR TITLE
Enable linting in the containers directory

### PR DESCRIPTION
**Pull request by m-czernek**
## What does this PR change?

In this PR, I am including Alex's formatting commit from https://github.com/agraul/uyuni/commit/1c2a1ab4fdefbc9cfc04c0411d3f7e7861e30c52 and adding additional changes, mostly to `tftp_wrapper.py`.

In this PR, we are enabling linting enforcement check on the `containers` directory. 

## Pending issues

Please add the missing docstrings in your classes. Notably, the `tftp_wrapper.py` file is missing docstrings. There are `TODO` comments instead.

## Suggested functional changes

Pay attention to CLI arguments for `tftp_wrapper`, notably: `--httpHost`, `--serverFqdn`, and `--proxyFqdn`. 
The arguments stayed the same, however, they are now stored in variables that conform to Python naming.
In addition to the internal changes, please consider changing the arguments to use the `--http-host`, `--server-fqdn`, and `--proxy-fqdn` naming, which is I believe standard for scripts.

Consider changing the `contentFilter` method name. I did not refactor the method name for fear of breaking things, but if you can, please modify it to Python's snake case convention: `content_filter`.

You have a lot of  `uyuni-configure` scripts. That name should be `uyuni_configure` according to https://peps.python.org/pep-0008/#package-and-module-names - consider renaming your scripts if possible.

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: containers/proxy-httpd-image/uyuni-configure.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 #!/usr/bin/python3
+# pylint: disable=invalid-name
+"""Configure script for Uyuni proxy container."""
```
Comment (cbosdo) [inline containers/proxy-httpd-image/uyuni-configure.py:3]:
```suggestion
"""Configure script for Uyuni proxy httpd container."""
```

File: containers/proxy-salt-broker-image/uyuni-configure.py [hunk 2]
```diff
@@ -1,22 +1,23 @@
 #!/usr/bin/python3
-
+# pylint: disable=invalid-name
+"""Configure script for Uyuni proxy Salt broker."""
```
Comment (cbosdo) [inline containers/proxy-salt-broker-image/uyuni-configure.py:3]:
```suggestion
"""Configure script for Uyuni proxy Salt broker container."""
```

File: containers/proxy-squid-image/uyuni-configure.py [hunk 3]
```diff
@@ -1,22 +1,30 @@
 #!/usr/bin/python3
+# pylint: disable=invalid-name
+"""Configure script for Uyuni proxy squid."""
```
Comment (cbosdo) [inline containers/proxy-squid-image/uyuni-configure.py:3]:
```suggestion
"""Configure script for Uyuni proxy squid container."""
```

File: containers/proxy-ssh-image/uyuni-configure.py [hunk 4]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/python3
+# pylint: disable=invalid-name
+"""Configure script for Uyuni proxy SSH."""
```
Comment (cbosdo) [inline containers/proxy-ssh-image/uyuni-configure.py:3]:
```suggestion
"""Configure script for Uyuni proxy SSH container."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 5]
```diff
@@ -2,6 +2,7 @@
 # SPDX-FileCopyrightText: 2024 SUSE LLC
 #
 # SPDX-License-Identifier: MIT
+"""Configure script for Uyuni proxy tftp."""
```
Comment (cbosdo) [inline containers/proxy-tftpd-image/tftp_wrapper.py:5]:
```suggestion
"""Wrapper script for Uyuni proxy tftp container."""
```

File: containers/proxy-tftpd-image/uyuni-configure.py [hunk 6]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/python3
+# pylint: disable=invalid-name
+"""Configure script for proxy TFTP"""
```
Comment (cbosdo) [inline containers/proxy-tftpd-image/uyuni-configure.py:3]:
```suggestion
"""Configure script for proxy TFTP container."""
```

**Review by cbosdo — APPROVED** [review]

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 7]
```diff
@@ -92,11 +97,15 @@ def __init__(self, url, capath, proxyFqdn, serverFqdn):
         self.contentFilter()
         self._size = len(self._content)
 
+    # pylint: disable-next=invalid-name
     def contentFilter(self):
         raise NotImplementedError()
 
 
 class HttpResponseDataFilteredPXE(HttpResponseDataFiltered):
+    """TODO: Add class docstring"""
```
Comment (aaannz) [inline containers/proxy-tftpd-image/tftp_wrapper.py:106]:
```suggestion
    """Filter pxelinux.cfg files for proxy instance."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 8]
```diff
@@ -143,6 +152,9 @@ def contentFilter(self):
 
 
 class HttpResponseDataFilteredGrub(HttpResponseDataFiltered):
+    """TODO: Add class dosctring"""
```
Comment (aaannz) [inline containers/proxy-tftpd-image/tftp_wrapper.py:155]:
```suggestion
    """Filter grub config files for proxy instance."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 9]
```diff
@@ -183,87 +195,91 @@ def contentFilter(self):
 
 
 class TFTPHandler(BaseHandler):
+    """TODO: Add class docstring"""
```
Comment (aaannz) [inline containers/proxy-tftpd-image/tftp_wrapper.py:198]:
```suggestion
    """Individual TFTP connection handler."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 10]
```diff
@@ -183,87 +195,91 @@ def contentFilter(self):
...
+                f"{target}/tftp/{path}", capath, self._proxy_fqdn, self._server_fqdn
             )
         # The rest get from http
-        logging.debug(f"Got request for {path}, forwarding to HTTP")
+        logging.debug("Got request for %s, forwarding to HTTP", path)
         return HttpResponseData(f"{target}/tftp/{path}", capath)
 
 
 class TFTPServer(BaseServer):
+    """TODO: Add class docstring"""
```
Comment (aaannz) [inline containers/proxy-tftpd-image/tftp_wrapper.py:259]:
```suggestion
    """Main TFTP server class."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 11]
```diff
@@ -35,6 +37,8 @@ def close(self):
 
 
 class HttpResponseData(ResponseData):
+    """TODO: Add class docstring"""
```
Comment (aaannz) [inline containers/proxy-tftpd-image/tftp_wrapper.py:40]:
```suggestion
    """Unprocessed HTTP response."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 12]
```diff
@@ -11,15 +12,16 @@
...
 from fbtftp.base_server import BaseServer
 
 
 def stats(s):
-    pass
+    del s
 
 
 class FileResponseData(ResponseData):
+    """TODO: Add class docstring"""
```
Comment (aaannz) [inline containers/proxy-tftpd-image/tftp_wrapper.py:23]:
```suggestion
    """Local file response."""
```

File: containers/proxy-tftpd-image/tftp_wrapper.py [hunk 13]
```diff
@@ -78,10 +82,11 @@ class HttpResponseDataFiltered(HttpResponseData):
     Saltboot entries have MASTER= option, we want only those mathing our proxy
     """
 
-    def __init__(self, url, capath, proxyFqdn, serverFqdn):
+    def __init__(self, url, capath, proxy_fqdn, server_fqdn):
+        super().__init__(url, capath)
```
[the remark this answers is not in the document]
Comment (m-czernek) [reply containers/proxy-tftpd-image/tftp_wrapper.py:86]: In Python, superclass's init is not called by default. Does this subclass not rely on the superclass's init? If not, we can remove the code and skip this check, but typically, this is an error by omission (https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/super-init-not-called.html)

[the remark this answers is not in the document]
Comment (m-czernek) [reply containers/proxy-tftpd-image/tftp_wrapper.py]: After taking a closer look at the two init methods, you're right, I don't think this call is correct. Ideally, we'd fix the messy inits too, but that's not in scope. I've removed the call and added a comment for pylint to ignore it. Thanks for noticing this Ondrej!

**Review by aaannz — APPROVED** [review]
LGTM, thanks.

One additional question; why are all `f"{var}"` strings replaced with their `"%s", var` variant in the tftp wrapper, but for example uyuni-configure in http image still uses `f""` variant?

**Comment by m-czernek** [comment]
@aaannz f-strings are preferred generally because they are easy to read. However, in logging functions, we prefer lazily-evaluated strings (since the log might never print) and f-strings evaluate eagerly [0] [1]. 

[0] https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/logging-fstring-interpolation.html
[1] https://google.github.io/styleguide/pyguide.html#3101-logging

**Pull request merged**